src/transport/SecureSessionTable.cpp

```diff
@@ -183,6 +220,100 @@ SecureSession * SecureSessionTable::EvictAndAllocate(uint16_t localSessionId, Se
 
 void SecureSessionTable::DefaultEvictionPolicy(EvictionPolicyContext & evictionContext)
 {
+    evictionContext.Sort([&evictionContext](const SortableSession & a, const SortableSession & b) {
+        //
+        // Sorting on Key1
+        //
+        if (a.mNumMatchingOnFabric > b.mNumMatchingOnFabric)
+        {
+            return true;
+        }
+
+        if (a.mNumMatchingOnFabric == b.mNumMatchingOnFabric)
+        {
+            int ScoreA = a.mSession->GetPeer().GetFabricIndex() == evictionContext.GetSessionEvictionHint().GetFabricIndex();
+            int ScoreB = b.mSession->GetPeer().GetFabricIndex() == evictionContext.GetSessionEvictionHint().GetFabricIndex();
+
+            //
+            // Sorting on Key2
+            //
+            if (ScoreA > ScoreB)
+            {
+                return true;
+            }
+
+            if (ScoreA == ScoreB)
+            {
+                //
+                // Sorting on Key3
+                //
+                if (a.mNumMatchingOnPeer > b.mNumMatchingOnPeer)
+                {
+                    return true;
+                }
+
+                if (a.mNumMatchingOnPeer == b.mNumMatchingOnPeer)
+                {
+                    int ScoreA1 = a.mSession->GetPeer() == evictionContext.GetSessionEvictionHint();
+                    int ScoreB1 = b.mSession->GetPeer() == evictionContext.GetSessionEvictionHint();
+
+                    //
+                    // Sorting on Key4
+                    //
+                    if (ScoreA1 > ScoreB1)
+                    {
+                        return true;
+                    }
+
+                    if (ScoreA1 == ScoreB1)
+                    {
+                        int aStateScore = 0, bStateScore = 0;
+
+                        auto assignStateScore = [](auto & score, const auto & session) {
```

Comment:
please extract as a static method. Maybe use an enum backed by int to make things readable like kDefunct, kActive, ...

Reply:
How exactly would you envision the enum helping here?